### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by shella12**
Created following pages

A login page.
A registration page.
A food list (with all CRUD implementation, except for 'update').
A recipes list (with all CRUD implementation, except for 'update').
A general shopping list view (all missing food for all your recipes and total price).
A list of all public recipes from other users with their names and total prices.

View is based on the provided wireframe:

Screenshot 2023-04-08 at 9 43 17 PM

### CODE REVIEW

File: app/controllers/foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+class FoodsController < ApplicationController
+  def index
+    @user = User.find(params[:user_id])
+    # @user = User.first
+    @foods = Food.find_by(user_id: params[:id])
+  end
+
```
Comment (ichala) [inline app/controllers/foods_controller.rb:7]: - [ ] to keep your code clean , kindly remove any commented code from your project 🙏

File: db/seeds.rb [hunk 2]
```diff
@@ -5,3 +5,18 @@
...
+
+recipe1 = Recipe.create(name: "Banana Smoothie", preparationTime: 10, cookingTime: 0, description: "A delicious smoothie made with banana, milk and honey", public: true, user_id: 1)
+recipe2 = Recipe.create(name: "Apple Pie", preparationTime: 30, cookingTime: 60, description: "A delicious pie made with apple, flour, sugar and butter", public: true, user_id: 1)
+recipe3 = Recipe.create(name: "Orange Juice", preparationTime: 10, cookingTime: 0, description: "A delicious juice made with orange, sugar and water", public: true, user_id: 1)
+
+recipe_food = RecipeFood.create(quantity: 1, recipe_id: recipe1.id, food_id: banana.id)
+recipe_food2 = RecipeFood.create(quantity: 1, recipe_id: recipe2.id, food_id: apple.id)
+recipe_food3 = RecipeFood.create(quantity: 1, recipe_id: recipe3.id, food_id: orange.id)
+recipe_food4 = RecipeFood.create(quantity: 1, recipe_id: recipe1.id, food_id: apple.id)
+recipe_food5 = RecipeFood.create(quantity: 100, recipe_id: recipe2.id, food_id: onion.id)
```
Comment (ichala) [inline db/seeds.rb:21]: - [ ] there's an error while seeding the database , the variables banana , apple , orange etc ... are not defined ! kindly fix that issue  
image

File: spec/controllers/shopping_list_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        get 'index', params: { user_id: user.id, recipe_id: recipe.id }
+        expect(response).to have_http_status(200)
+      end
+
+      it 'should have test: "Shopping List"' do
+        get 'index', params: { user_id: user.id, recipe_id: recipe.id }
+        expect(response.body).to include('Shopping List')
+      end
+    end
+  end
```
Comment (ichala) [inline spec/controllers/shopping_list_controller_spec.rb:24]: - [ ] you have some failing tests in your application , kindly fix it 🙏 

image

File: app/controllers/recipes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  def toggle
+    @recipe = Recipe.find(params[:id])
+    @recipe.toggle!(:public)
+    redirect_to recipes_detail_path(params[:id])
+  end
+
+  def public_recipes
+    @recipes = Recipe.all.where(public: true)
+  end
+end
```
Comment (ichala) [inline app/controllers/recipes_controller.rb:25]: - [ ] Please note that the recipe & public recipe pages are returning an empty page with no button to create a recipe, kindly fix that issue by adding a link to create a recipe and make sure it matches the wireframe and has user authorization rules 🙏 


image

**Review by ichala — CHANGES_REQUESTED** [review]
Hi Team
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No Linter errors  ✔️ 
- Device used ✔️


 ## Required Changes ♻️

- [ ] **_🛑  Check The Comments Below_**





## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/recipes_controller.rb (hunk 4 above)
Comment (PinkMoon25) [inline app/controllers/recipes_controller.rb:25]: - [ ] As requested by the previous reviewer, please make sure there is a way to create recipes, I think you are missing the create recipe method in the RecipesController, and also missing the form to create recipe in the views, so just like you have it for the FoodsController, please make sure the user is able to create recipes :+1: 

currently we get an empty view with no action that a user can take on the `/recipes` path please see :point_down: 
image

according to the wireframe /recipes path should display the recipes belonging to the user along with a button or link to create a recipe which will take us to the form to create a recipe, and then clicking on each recipe will take us to that particular recipes detail page showing the details of the recipe and an option to generate shopping list like this :point_down: 
image

**Review by pinkmoon25 — CHANGES_REQUESTED** [review]
Hi @shella12 @ZuhRa-HashiMi , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Rubocop checks :heavy_check_mark: 
gitflow :heavy_check_mark: 
Pair-programming :heavy_check_mark: 
Food section :heavy_check_mark: 
Rspecs :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Owusu-Desmond — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Great working on the setup of the project ✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**